catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/RestoreCommand.java

```diff
@@ -0,0 +1,318 @@
+      throws IOException, SolrServerException {
+
+    if (canRestore(client, collection)) {
+      CollectionAdminRequest.Restore restore =
+          CollectionAdminRequest.AsyncCollectionAdminRequest.restoreCollection(
+                  collection, backupName)
+              .setLocation(backupLocation);
+
+      String requestId = restore.processAsync(client);
+      LOGGER.debug("Restore request Id: {}", requestId);
+      return requestId;
+    } else {
+      printErrorMessage(
+          String.format("Unable to restore %s, collection already exists.", collection));
+      LOGGER.debug("Unable to restore: {}, collection already exists", collection);
+      return null;
+    }
+  }
+
+  private void performSolrCloudRestore(SolrClient client) {
+    LOGGER.debug("Restoring collection {} from {} / {}.", coreName, backupLocation, backupName);
+    printInfoMessage(
+        String.format(
+            "Restoring collection [%s] from [%s] / [%s].", coreName, backupLocation, backupName));
+    try {
+      if (asyncRestore) {
+        String requestId = restoreAsync(client, coreName, backupLocation, backupName);
+        printInfoMessage("Restore request Id: " + requestId);
+      } else {
+        boolean isSuccess = restore(client, coreName, backupLocation, backupName);
+        if (isSuccess) {
+          printInfoMessage("Restore complete.");
+        }
+      }
+    } catch (Exception e) {
+      String message = e.getMessage() != null ? e.getMessage() : "";
+      printErrorMessage(String.format("Restore failed. %s", message));
+      LOGGER.debug("Restore failed for core: {}.", coreName, e);
+    }
+  }
+
+  private boolean canRestore(SolrClient client, String collection)
+      throws IOException, SolrServerException {
+
+    LOGGER.trace("Checking restoration capability of collection {}", collection);
+    if (collectionExists(client, collection)) {
+      LOGGER.trace("Collection {} already exists", collection);
+      if (force) {
+        LOGGER.trace("Force option set, deleting existing {} collection", collection);
+        CollectionAdminRequest.Delete delete = CollectionAdminRequest.deleteCollection(collection);
+        CollectionAdminResponse response = delete.process(client);
+        return response.isSuccess();
+      } else {
+        printErrorMessage(
+            String.format(
+                "Force option not set, cannot restore %s collection. Use --force to restore when the collection exists.",
+                collection));
+        LOGGER.trace("Force option not set, cannot restore {} collection", collection);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private void createSolrCore() throws IOException, SolrServerException {
+    httpBuilder = new org.codice.solr.factory.impl.HttpClientBuilder(encryptionService);
+    String url = HttpSolrClientFactory.getDefaultHttpsAddress();
+    final String solrDataDir = HttpSolrClientFactory.getSolrDataDir();
+    if (solrDataDir != null) {
```

Comment:
The `createSolrCore` method checks if the Solr data dir is null. I think the check on line 283 might not be necessary.